microcosm_daemon/standby.py

```diff
@@ -0,0 +1,77 @@
+"""
+Standby state machine.
+
+"""
+from abc import ABCMeta, abstractproperty
+from six import add_metaclass
+
+from microcosm_daemon.sleep_policy import SleepNow
+
+
+class StandByGuard(object):
+    """
+    State wrapper for a standby-enabled daemon.
+
+    Calls state and then checks condition.
+
+    """
+    def __init__(self, next_state, condition):
+        self.next_state = next_state
+        self.condition = condition
+
+    def __call__(self, graph):
+        """
+        Invoke the wrapped state and then check the standby condition.
+
+        """
+        result = self.next_state(graph)
+
+        should_standby = self.condition(graph)
+        if should_standby:
+            return StandByState(result or self.next_state, self.condition)
+
+        return StandByGuard(result, self.condition)
+
+
+class StandByState(object):
```

Comment:
We need a way to avoid running the normal state machine; this state just checks the condition and sleeps.